File: problems/220/solution.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @param {number} t
 * @return {boolean}
 */
var containsNearbyAlmostDuplicate = function(nums, k, t) {
    for (let i = 0; i < nums.length; i++) {
        for (let j = Math.max(0, i - k); j < i; j++) {
            if (Math.abs(nums[i] - nums[j]) <= t) {
                return true;
            }
        }
    }
    
    return false;
};

File: problems/220/solution.go
package main
func containsNearbyAlmostDuplicate(nums []int, k int, t int) bool {
    // sort.Ints(nums)
    
    for i := 0; i < len(nums) - 1; i++ {
        for j := i + 1; j < len(nums); j++ {
            if abs(j - i) <= k && abs(nums[j] - nums[i]) <= t {
                return true
            }
        }
    }
    
    return false
}

func abs(num int) int {
    if num < 0 {
        return num * -1
    }
    
    return num
}